### PR TITLE
CATROID-275 fix cloneCompositeBrickUserVariableBug

### PR DESCRIPTION
**Pull request by gPathpp**

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/CloneCompositeBrickUpdateVariableTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+	@Test
+	public void testIf() {
+		IfLogicBeginBrick ifBrick = new IfLogicBeginBrick();
+		ifBrick.addBrickToIfBranch(brick);
+		script.addBrick(ifBrick);
+
+		Sprite cloneSprite = new SpriteController().copyForCloneBrick(sprite);
+		Brick cloneIfLogicBrick = cloneSprite.getScript(0).getBrick(0);
+		checkInNestedBrick((CompositeBrick) cloneIfLogicBrick);
```
Comment (dpedratscher) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/CloneCompositeBrickUpdateVariableTest.java]: the function name could be more expressive. It's not clear, what you check. maybe checkCloneForNestedBrick or something better?

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/ScriptController.java [hunk 2]
```diff
@@ -207,4 +209,23 @@ void unpackForSprite(Script scriptToUnpack, Project dstProject, Scene dstScene,
 
 		dstSprite.getScriptList().add(script);
 	}
+
+	private List<Brick> getAllBricksFromScriptBrickList(List<Brick> scriptBrickList) {
```
Comment (dpedratscher) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/ScriptController.java]: could you please add a test for this function?

**Comment by wslany** [comment]
Thanks!!

File: catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/CloneCompositeBrickUpdateVariableTest.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+		checkClonedBrickInNestedBrick((CompositeBrick) cloneRepeatBrick1);
+	}
+
+	private void checkClonedBrickInNestedBrick(CompositeBrick compositeBrick) {
+		Brick nestedBrick = compositeBrick.getNestedBricks().size() == 0
+				? compositeBrick.getSecondaryNestedBricks().get(0)
+				: compositeBrick.getNestedBricks().get(0);
+
+		if (nestedBrick instanceof UserVariableBrick) {
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/CloneCompositeBrickUpdateVariableTest.java]: please put this as a new test method into TwoPartCompositeBrickTreeBuilderTest and ThreePart....

**Pull request merged**